tests/test_plugins.py

```diff
@@ -1,46 +1,54 @@
-from cms.api import add_plugin, create_page
+from cms.api import add_plugin, create_page, create_title
 from cms.test_utils.testcases import CMSTestCase
 
-from djangocms_snippet.models import Snippet
+from .utils.factories import (
+    SnippetWithVersionFactory,
+)
 
 
 class SnippetPluginsTestCase(CMSTestCase):
 
     def setUp(self):
         self.language = "en"
-        self.home = create_page(
-            title="home",
-            template="page.html",
-            language=self.language,
-        )
-        self.page = create_page(
-            title="help",
-            template="page.html",
-            language=self.language,
-        )
-        self.pagecontent = self.page.pagecontent_set.first()
-        self.placeholder = self.pagecontent.placeholders.get(slot="content")
         self.superuser = self.get_superuser()
-
-    def tearDown(self):
-        self.page.delete()
-        self.home.delete()
-        self.superuser.delete()
+        page_data = {
```

Comment:
We should try and change as little as possible of the existing code, this makes maintenance easier. The master branch will receive updates for v3 and we will need to port those updates back in, if there is more noise than needed this can make the job harder. 
Can you please move this back to match the original and only change what you need to change, Also the api published param is deprcated and will throw a warning:
```
    self.home = create_page(
            title="home",
            template="page.html",
            language=self.language,
            "created_by": self.superuser
        )
```